### DISCUSSION TITLE
Parabolic SAR (stop and reverse)

### DISCUSSION THREAD

**Discussion by DaveSkender** [DaveSkender/Stock.Indicators#245] [Indicators]
Community discussion about [Parabolic SAR (stop and reverse)](https://dotnet.stockindicators.dev/indicators/ParabolicSar/#content)

**Comment by Salardx** [comment]
Hi dave, thank you for your tools
I utlilized this indicator and I noticed sometimes its sort of repainting
When the candles is settled and a new candle emerged, is it possible that the value of SAR indicator would change for the settled candle?

**Reply by DaveSkender** [reply]
The formula for Parabolic SAR does not repaint -- each bar only uses current and prior bar values.  With that said, if your underlying quote data is changed or corrected between subsequent calls or if you're using a non-static parameter argument then it could repaint the entire series.  Ref: [the code](https://github.com/DaveSkender/Stock.Indicators/blob/57cd86dfd5fb8b18302f96e5eb228122f5299f49/src/m-r/ParabolicSar/ParabolicSar.cs).

**Reply by DaveSkender** [reply]
Another scenario that I need to check is when you’re providing a fixed length quote history with insufficient length that does not get to past the first pivot, it should not produce any results.  If it is, that first leg would repaint and should not be used.  Testing here: #743

**Reply by DaveSkender** [reply]
After testing, there is an edge case where repainting occurred — when there is no initial reversal (not enough data or change in price).  We resolved this in #758 to exclude the offending values in this scenario.

**Comment by DaveSkender** [comment]
If you've found this page and are seeking to understand why our indicator does not match TradingView, please refer to these prior analysis.

- #1031 
- #801
